components/compliance-service/api/reporting/server/server.go

```diff
@@ -305,11 +305,16 @@ func csvExport(stream reporting.ReportingService_ExportServer) exportHandler {
 
 // ListNodes returns a list of nodes based on query
 func (srv *Server) ListNodes(ctx context.Context, in *reporting.Query) (*reporting.Nodes, error) {
+	formattedFilters := formatFilters(in.Filters)
+	var platform_sort_field = "platform.name.lower"
+	if len(formattedFilters["platform"]) == 1 {
+		platform_sort_field = "platform.release.lower"
+	}
 	var nodes reporting.Nodes
 	var SORT_FIELDS = map[string]string{
 		"name":                                   "node_name.lower",
 		"environment":                            "environment.lower",
-		"platform":                               "platform.name.lower",
+		"platform":                               platform_sort_field,
```

Comment:
I think the logic above is unnecessary, I think you could get the same effect by just changing this line to

```
"platform":                               "platform.release.lower",
```

Reply:
I was tried the same changes but in case of multiple platforms this changes not working because our requirement is only for in case of single platform, in case of multiple platforms we need to sort according to name only.

Reply:
> in case of multiple platforms we need to sort according to name only.

Why would it be better to sort by platform name instead of platform name and platform version? If multiple platforms are selected, sorting by platform name is just as wrong as in all the other cases.

Reply:
To be clear, I don't think it is correct to ever sort by just platform name. If I have nodes like this:

| id | platform name | platform version |
| -- | -- | -- | 
| 1 | ubuntu | 1 |
| 2 | ubuntu | 2 |
| 3 | centos | 1 |
| 4 | centos | 2 |

The correct order for sorting by platform would result in the order [3,4,1,2] displaying "centos 1", "centos 2", "ubuntu 1", "ubuntu 2". If they are only sorted by platform name, the centos nodes will appear before the ubuntu nodes, but the centos and ubuntu nodes will be sorted in arbitrary order.

Reply:
```
"platform":                               "platform.release.lower",
```
if I add changes like above and If I do not add any platform filter then sort by behavior is not working properly, if you have any suggestion on that please mentioned here for the same line of code. 
I also tried changes like
```
"platform.name":                          	  "platform.name.lower",
"platform.release":                          	  "platform.release.lower",
```

Reply:
We probably do not want to change the `"platform"` value in the left column, that is coming from the UI and changing it would probably require a more significant update.

We already have this code that adds a secondary sort for node_name: https://github.com/chef/automate/blob/462bfdca321c8926017ee5f48f46dc3ba927f891/components/compliance-service/reporting/relaxting/nodes.go#L83-L86

What if you added a second condition above that like:

```
        // Sorting by platform requires sorting by platform name and version
	if sortField == "platform.name.lower" {
		searchSource.Sort("platform.release.lower", sortAsc)
	}
```

This couples `GetNodes` and `ListNodes` but they are already coupled and this does not seem too bad.

Reply:
I tried with the changes
```
        if sortField == "platform.name.lower" {
		searchSource.Sort("platform.release.lower", sortAsc)
	}
```
but not worked.

Reply:
It didn't work at all, meaning it didn't return the proper results? Or something else? I wonder if our existing test data would be enough to reproduce this in a regression test.

Reply:
hi @phiggins have you tested with the below code changes because of these changes not work as expected. 
```
        if sortField == "platform.name.lower" {
		searchSource.Sort("platform.release.lower", sortAsc)
	}
```
actually, the ZD card requirement was only for the single platform, that's the reason I had did the same changes in A1, can we use these changes for now or if you want to sort multiple platforms same as your above example then we can discuss it more.

Reply:
@vinay033 can you link me to the ZD card? i want to make sure it aligns with the ux design direction from @jonong1972.

Reply:
found it: https://chefio.atlassian.net/browse/MSYS-828

Reply:
I have not tested this. I haven't spent the time to get setup with all the required data. The problem is that the existing code does not handle sorting by platform properly at all, regardless of whatever special case is applied.

Ideally we would have a test case for this but I know getting test data setup is not straightforward so I don't know how easily doable this is. If it is this hard to demonstrate what the problem is, it is going to be hard to verify this isn't broken in the future.